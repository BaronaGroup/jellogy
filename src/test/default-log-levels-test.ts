import {prepareTestKarhu} from './test-utils'

describe('default-log-levels-test', () => {
  const karhuTest = prepareTestKarhu({ defaultLogLevel: 'DEBUG'})

  it('logs things', karhuTest((karhu, output) => {
    const context = karhu.context('test-context')
    context.debug('dbg')
    context.info('inf')
    context.warn('wrn')
    context.error('err')
    expect(output.tracked).toMatchSnapshot()
  }))
})
