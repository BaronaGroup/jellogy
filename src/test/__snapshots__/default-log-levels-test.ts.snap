// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-log-levels-test logs things 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [
    Array [
      "[1] [95mDEBUG test-context dbg[39m",
      "DEBUG",
      "test-context",
    ],
  ],
  "INFO" => Array [
    Array [
      "[2] INFO test-context inf",
      "INFO",
      "test-context",
    ],
  ],
  "WARN" => Array [
    Array [
      "[3] [93mWARN test-context wrn[39m",
      "WARN",
      "test-context",
    ],
  ],
  "ERROR" => Array [
    Array [
      "[4] [97m[101mERROR test-context err[49m[39m",
      "ERROR",
      "test-context",
    ],
  ],
}
`;
